### PR TITLE
fix requirements + aggregate setup & req. & config in pyproject

### PR DESCRIPTION
**Pull request by Borda**
- fixed requirements/version so it can be run (needed after the latest Torch release)
- update to standard Python project configuration and as an advantage it allows us to add optional settings for pytest/back/etc. without adding a new file

### CODE REVIEW

File: .github/azure-gpu-tests.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
         python -c "import torch ; mgpu = torch.cuda.device_count() ; assert mgpu == 2, f'GPU: {mgpu}'"
       displayName: 'Image info & NVIDIA'
 
-    - script: pip install pytest -r requirements.txt
+    - script: pip install .[all] pytest
```
Comment (rasbt) [inline .github/azure-gpu-tests.yml:53]: Does this need to be `pip install ".[all]" pytest` (I guess it depends on the shell)

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+[project]
+name = "lit-llama"
+version = "0.1.0"
+description = "Implementation of the LLaMA language model"
+license = {text = "Apache-2.0"}
+authors = [
+    { name = "Lightning AI", email = "community@lightning.ai" }
+]
+readme = "README.md"
+requires-python = ">=3.8"
```
Comment (rasbt) [inline pyproject.toml:14]: In LitGPT we require >= 3.10 because we used some features not supported in 3.9 (I don't remember which ones); so maybe also make this 3.10 here to be on the safe side

**Review by rasbt — APPROVED** [review]

File: .github/azure-gpu-tests.yml (hunk 1 above)
Comment (Borda) [reply .github/azure-gpu-tests.yml:53]: Linux not but Mac yes... anyway we can have it :)
```suggestion
    - script: pip install ".[all]" "pytest"
```

File: pyproject.toml (hunk 2 above)
Comment (Borda) [reply pyproject.toml:14]:
```suggestion
requires-python = ">=3.10"
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    "bitsandbytes",
+]
+classifiers = [
+    "Topic :: Text Processing"
+]
+
+[project.optional-dependencies]
+all = [
+    "tqdm",  # convert_checkpoint.py
+    "numpy",  # train.py dataset memmap
```
Comment (Borda) [inline pyproject.toml:28]:
```suggestion
    "numpy <2.0",  # train.py dataset memmap
```
